### PR TITLE
185274315 esg funding report

### PR DESCRIPTION
**Pull request by ZachCMP**
https://www.pivotaltracker.com/story/show/185274315

Relies on backend PR: https://github.com/greenriver/hmis-warehouse/pull/3204

### CODE REVIEW

File: src/api/operations/reports.fragments.graphql [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    repeats
+    value {
+      id
+      valueBoolean
+      valueDate
+      valueFloat
+      valueInteger
+      valueJson
+      valueString
+      valueText
```
Comment (gigxz) [inline src/api/operations/reports.fragments.graphql:33]:
```suggestion
      ...CustomDataElementFields
```

File: src/api/operations/reports.fragments.graphql [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    }
+  }
```
Comment (gigxz) [inline src/api/operations/reports.fragments.graphql]:
```suggestion
  customDataElements {
    ...CustomDataElementFields
  }
```

File: src/api/operations/reports.fragments.graphql [hunk 3]
```diff
@@ -0,0 +1,46 @@
+fragment EsgFundingServiceFields on EsgFundingService {
+  id
+  clientId
+  clientDob
+  mciIds {
+    id
+    identifier
+    label
+    url
```
Comment (gigxz) [inline src/api/operations/reports.fragments.graphql]:
```suggestion
    ...ClientIdentifierFields
```

File: src/modules/projects/components/EsgFundingReport.tsx [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+          header: 'First Name',
+          render: (row) => row.firstName,
+        },
+        {
+          header: 'Date Of Birth',
+          render: (row) => parseAndFormatDate(row.clientDob),
+        },
+        {
+          header: 'MCI_ID',
+          render: (row) => row.mciIds.map((id) => id.identifier).join('\n'),
```
Comment (gigxz) [inline src/modules/projects/components/EsgFundingReport.tsx:87]:
```suggestion
          header: 'MCI ID',
          render: (row) => (<ExternalIdDisplay
        value={row.mciIds}
      />),
```
That should make the links show up too (untested)

File: src/modules/projects/components/EsgFundingReport.tsx [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+        {
+          header: 'First Name',
+          render: (row) => row.firstName,
+        },
+        {
+          header: 'Date Of Birth',
+          render: (row) => parseAndFormatDate(row.clientDob),
+        },
+        {
+          header: 'MCI_ID',
```
Comment (gigxz) [inline src/modules/projects/components/EsgFundingReport.tsx]:
```suggestion
          header: 'MCI ID',
```

File: src/modules/projects/components/EsgFundingReport.tsx [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+        },
+        {
+          header: 'Program Name',
+          render: (row) => row.projectName,
+        },
+        {
+          header: 'Funding Source',
+          render: (row) =>
+            row.customDataElements?.find((e) => e.key === 'funding_source')
+              ?.value?.valueString,
```
Comment (gigxz) [inline src/modules/projects/components/EsgFundingReport.tsx]:
```suggestion
            customDataElementValueForKey('funding_source', row)
```

File: src/modules/projects/components/EsgFundingReport.tsx [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+        },
+        {
+          header: 'Payment Type',
+          render: (row) =>
+            row.customDataElements?.find((e) => e.key === 'payment_type')?.value
+              ?.valueString,
```
Comment (gigxz) [inline src/modules/projects/components/EsgFundingReport.tsx]:
```suggestion
            customDataElementValueForKey('payment_type', row)
```

File: src/modules/projects/components/EsgFundingReport.tsx [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+  EsgFundingServiceFieldsFragment,
+  useGetEsgFundingReportQuery,
+  useGetReferralPostingQuery,
+} from '@/types/gqlTypes';
+
+const EsgFundingReport: React.FC = () => {
+  const { referralPostingId } = useSafeParams<{ referralPostingId: string }>();
+  const { data, loading, error } = useGetReferralPostingQuery({
+    variables: { id: referralPostingId as any as string },
+    fetchPolicy: 'network-only',
```
Comment (gigxz) [inline src/modules/projects/components/EsgFundingReport.tsx]: Why use network only? We definitely have it in the cache already when navigating to the ESG funding report, so I don't think we want to refetch it.

**Review by gigxz — COMMENTED** [review]
I did an initial review but I'm unable to test it due to an error (see screenshot on the backend pr). Code looks good though, some minor comments inline.

File: src/modules/projects/components/EsgFundingReport.tsx (hunk 8 above)
Comment (ZachCMP) [reply src/modules/projects/components/EsgFundingReport.tsx]: Sounds good, I'll switch it

**Comment by ZachCMP** [comment]
@gigxz All the things should be fixed now

**Review by gigxz — APPROVED** [review]

**Pull request merged**